doc/api/crypto.md

```diff
@@ -1046,6 +1046,76 @@ encoding of `'utf8'` is enforced. If `data` is a [`Buffer`][], `TypedArray`, or
 
 This can be called many times with new data as it is streamed.
 
+## Class: KeyObject
+<!-- YAML
+added: REPLACEME
+-->
+
+Node.js uses an internal `KeyObject` class which should not be accessed
+directly. Instead, factory functions exist to create instances of this class
+in a secure manner, see [`crypto.createSecretKey`][],
+[`crypto.createPublicKey`][] and [`crypto.createPrivateKey`][]. A `KeyObject`
+can represent a symmetric or asymmetric key, and each kind of key exposes
+different functions.
+
+Most applications should consider using the new `KeyObject` API instead of
+passing keys as strings or `Buffer`s due to improved security features.
+
+### keyObject.getType()
```

Comment:
I considered that, and it would certainly have some advantages (e.g. retrieving `symmetricKeySize` would not throw an error on asymmetric keys (or should it?)), but most other crypto APIs use functions instead of getters. What do you think?

Reply:
I'd be +1 on using getters. It's a nicer API model, in my opinion. We shouldn't need to feel too constrained by the existing API pattern here.